mesonbuild/interpreter/interpreter.py

```diff
@@ -1531,6 +1531,10 @@ def add_languages_for(self, args: T.List[str], required: bool, for_machine: Mach
                         continue
                     else:
                         raise
+                if lang == 'cuda' and hasattr(self.backend, 'allow_thin_archives'):
+                    # see NinjaBackend.__init__() why we need to disable thin archives for cuda
+                    mlog.debug('added cuda as language, disabling thin archives for {} machine, since nvcc/nvlink cannot handle thin archives natively'.format(for_machine))
+                    self.backend.allow_thin_archives[for_machine] = False
```

Comment:
Ah right, we need this because the backend is initialized at the end of project() and we set it up in the backend's `__init__`.